src/librustdoc/html/static/js/search.js

```diff
@@ -1344,15 +1356,32 @@ function initSearch(rawSearchIndex) {
                 }
                 return false;
             };
+            /**
+             * @param {number} id
+             * @param {[QueryElement]} queryElemList
+             */
             const handleQueryElemList = (id, queryElemList) => {
-                if (!fnTypeSet.has(id)) {
-                    if (id === typeNameIdOfArrayOrSlice) {
+                let fsid = id;
+                if (fsid < 0) {
+                    fsid = null;
+                    if (!mgens) {
+                        return false;
+                    }
```

Comment:
```suggestion
                    if (!mgens) {
                        return false;
                    }
                    fsid = null;
```

Changing the order a bit. :)